aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -192,6 +192,7 @@ private RESTClient httpClient() {
               HTTPClient.builder(properties())
                   .uri(baseSignerUri())
                   .withObjectMapper(S3ObjectMapper.mapper())
+                  .withAuthSession(org.apache.iceberg.rest.auth.AuthSession.EMPTY)
```

Comment:
Why are we setting this to EMPTY?  In order to sign we need an auth session.  Wouldn't this result in no headers/sigs being produced if a call was made without explicitly setting the auth session?  We want enforce that an auth session is used, but this results in falling back to nothing.

Reply:
I agree with your suggestions for other components, but here I disagree: the HTTP client _must_ be configured with an empty auth session because it will be used for token refreshes. If it contains, say, token `t1`, and token `t1` gets refreshed and exchanged for `t2`, there would be a clash between the token from the HTTP client (`t1`) and the one from the refreshed auth session (`t2`).

Reply:
I will however move this call to the `authSession()` and add a comment to explain why an `EMPTY` session is needed in this specific case.